```
CMake Error at lib/CMakeLists.txt:58 (target_link_libraries):
  Cannot specify link libraries for target "gainput" which is not built by
  this project.
```

--- a/ports/gainput/install_as_cmake_package.patch
+++ b/ports/gainput/install_as_cmake_package.patch
@@ -1,7 +1,16 @@
 diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
-index 14d5e85..ce2f5a8 100644
+index 14d5e85..803dd3b 100644
 --- a/lib/CMakeLists.txt
 +++ b/lib/CMakeLists.txt
+@@ -21,7 +21,7 @@ if (GAINPUT_BUILD_SHARED)
+   add_library(gainput SHARED ${sources} ${mmsources})
+   set_target_properties(gainput PROPERTIES
+     OUTPUT_NAME gainput
+-    DEBUG_POSTFIX -d
++    DEBUG_POSTFIX d
+     VERSION ${GAINPUT_VERSION}
+     SOVERSION ${GAINPUT_MAJOR_VERSION}
+     FOLDER gainput
 @@ -30,19 +30,27 @@ if (GAINPUT_BUILD_SHARED)
  endif (GAINPUT_BUILD_SHARED)
  
@@ -9,7 +18,8 @@ index 14d5e85..ce2f5a8 100644
 -  message(STATUS "..Building shared libraries (-DGAINPUT_BUILD_STATIC=OFF to disable)")
 +  message(STATUS "..Building static libraries (-DGAINPUT_BUILD_STATIC=OFF to disable)")
    add_library(gainputstatic STATIC ${sources} ${mmsources})
-   set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX -d FOLDER gainput)
+-  set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX -d FOLDER gainput)
++  set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX d FOLDER gainput)
    set(install_libs ${install_libs} gainputstatic)
  endif (GAINPUT_BUILD_STATIC)
  
@@ -35,24 +45,7 @@ index 14d5e85..ce2f5a8 100644
  elseif(APPLE)
  	find_library(FOUNDATION Foundation)
  	find_library(IOKIT IOKit)
-@@ -52,10 +60,14 @@ elseif(APPLE)
-       find_library(UIKIT UIKit)
-       find_library(COREMOTION CoreMotion)
-       find_library(QUARTZCORE QuartzCore)
--      target_link_libraries(gainput ${UIKIT} ${COREMOTION})
-+      if (GAINPUT_BUILD_SHARED)
-+        target_link_libraries(gainput ${UIKIT} ${COREMOTION})
-+      endif(GAINPUT_BUILD_SHARED)
-   else()
-       find_library(APPKIT AppKit)
--      target_link_libraries(gainput ${APPKIT})
-+      if (GAINPUT_BUILD_SHARED)
-+        target_link_libraries(gainput ${APPKIT})
-+      endif(GAINPUT_BUILD_SHARED)
-   endif()
- endif()
- 
-@@ -65,6 +77,14 @@ if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+@@ -65,6 +73,14 @@ if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
  endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
  set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
  
@@ -67,7 +60,7 @@ index 14d5e85..ce2f5a8 100644
  install(
      DIRECTORY "include/gainput"
      DESTINATION "include"
-@@ -73,7 +93,14 @@ install(
+@@ -73,7 +89,15 @@ install(
  
  install(
      TARGETS ${install_libs}
@@ -76,6 +69,7 @@ index 14d5e85..ce2f5a8 100644
      ARCHIVE DESTINATION "${libdir}"
      RUNTIME DESTINATION "bin"
  )
++
 +install(
 +    EXPORT unofficial-${PROJECT_NAME}-Targets
 +    FILE unofficial-${PROJECT_NAME}Config.cmake

--- a/ports/gainput/install_as_cmake_package.patch
+++ b/ports/gainput/install_as_cmake_package.patch
@@ -1,16 +1,7 @@
 diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
-index 14d5e85..803dd3b 100644
+index 14d5e85..cb7232f 100644
 --- a/lib/CMakeLists.txt
 +++ b/lib/CMakeLists.txt
-@@ -21,7 +21,7 @@ if (GAINPUT_BUILD_SHARED)
-   add_library(gainput SHARED ${sources} ${mmsources})
-   set_target_properties(gainput PROPERTIES
-     OUTPUT_NAME gainput
--    DEBUG_POSTFIX -d
-+    DEBUG_POSTFIX d
-     VERSION ${GAINPUT_VERSION}
-     SOVERSION ${GAINPUT_MAJOR_VERSION}
-     FOLDER gainput
 @@ -30,19 +30,27 @@ if (GAINPUT_BUILD_SHARED)
  endif (GAINPUT_BUILD_SHARED)
  
@@ -18,8 +9,7 @@ index 14d5e85..803dd3b 100644
 -  message(STATUS "..Building shared libraries (-DGAINPUT_BUILD_STATIC=OFF to disable)")
 +  message(STATUS "..Building static libraries (-DGAINPUT_BUILD_STATIC=OFF to disable)")
    add_library(gainputstatic STATIC ${sources} ${mmsources})
--  set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX -d FOLDER gainput)
-+  set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX d FOLDER gainput)
+   set_target_properties(gainputstatic PROPERTIES DEBUG_POSTFIX -d FOLDER gainput)
    set(install_libs ${install_libs} gainputstatic)
  endif (GAINPUT_BUILD_STATIC)
  

--- a/ports/gainput/portfile.cmake
+++ b/ports/gainput/portfile.cmake
@@ -10,13 +10,13 @@ vcpkg_from_github(
         "install_as_cmake_package.patch"
 )
 
-set(GAINPUT_BUILD_SHARED ON)
-set(GAINPUT_BUILD_STATIC OFF)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     set(GAINPUT_BUILD_SHARED OFF)
     set(GAINPUT_BUILD_STATIC ON)
+else()
+    set(GAINPUT_BUILD_SHARED ON)
+    set(GAINPUT_BUILD_STATIC OFF)
 endif()
-
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}

--- a/ports/gainput/portfile.cmake
+++ b/ports/gainput/portfile.cmake
@@ -10,10 +10,9 @@ vcpkg_from_github(
         "install_as_cmake_package.patch"
 )
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
-    set(GAINPUT_BUILD_SHARED ON)
-    set(GAINPUT_BUILD_STATIC OFF)
-elseif(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+set(GAINPUT_BUILD_SHARED ON)
+set(GAINPUT_BUILD_STATIC OFF)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     set(GAINPUT_BUILD_SHARED OFF)
     set(GAINPUT_BUILD_STATIC ON)
 endif()

